doc/website/getting-started/overview.md

```diff
@@ -0,0 +1,540 @@
+    // handle the error
+}
+```
+
+Should we need an error value we set
+```
+result = iox::cxx::error<E>(errorCode);
+```
+which assumes that E can be constructed from an ``errorCode``.
+
+We again can employ a functional approach like this
+```
+auto handleValue = [](int& value) { /*do something with the value*/ };
+auto handleError = [](E& value) { /*handle the error*/ };
+result.and_then(handleValue).or_else(handleError);
+```
+
+There are more convenience functions such as ``value_or`` which provides the value or an alternative specified by the
+user. These can be found in ``expected.hpp``.
+
+
+## Using the API
+
+Armed with the terminology and functional concepts, we can start to use the API to send and receive data. This
+involves settting up the runtime in each application, creating publishers in applications that need to send data and
+subscribers in applications that want to receive said data. An application can have both, publishers and subscribers.
+It can even send data to itself, but this usually makes little sense.
+
+### Initialize the runtime
+
+Create a runtime with a unique name among all applications for each application
+
+```
+iox::runtime::PoshRuntime::initRuntime("some_unique_name");
+```
+
+Now this application is ready to communicate with the middleware daemon Roudi.
+
+### Defining a topic
+
+We need to define a data type we can send, which can be any struct or class or even a plain type, such as an int.
+
+```
+struct CounterTopic
+{
+    CounterTopic(uint32_t counter = 0U) : counter(counter) {}
+
+    uint32_t counter;
+};
+```
+
+The topic type must be default- and copy-constructible when the typed API is used. Using the untyped API imposes no
+such restriction, it just has to be possible to construct the data type at a given memory location.
+
+### Typed API
+
+We now demonstrate how to send and receive data with the typed API. 
+
+
+#### Creating a publisher
+
+We create a publisher that offers our CounterTopic.
+
+```
+iox::popo::TypedPublisher<CounterTopic> publisher({"Group", "Instance", "CounterTopic"});
+publisher.offer();
+```
+
+Note that it suffices to set the first two identifiers (Group and Instance) to some default values for all topics.
+#### Sending data
+
+Now we can use the publisher to send the data in various ways.
+
+1. Loan and assign
+
+    ```
+    auto result = publisher.loan();
+    if (!result.has_error())
+    {
+        auto& sample = result.value();
+        sample->counter = 73;
+        sample.publish();
+    } else 
+    {
+        //handle the error
+    }
+    ```
+
+    Here result is an ``expected`` and hence we may get an error which we have to handle. This can happen if we try
+    to loan to many samples and exhaust memory.
+    
+    If we successfully get a sample, we can use ``operator->`` to access the underlying data and set it to the value
+    we want to send. It is important to note that in the untyped case we get a default constructed topic and such an
```

Comment:
You meant ..we do **not** get.. @budrus What is the contract? untyped: loan does not call c'tor, typed: loan does call c'tor?

Reply:
No, the typo is that here I am talking about the typed case (not **un**typed). Corrected.